examples/inspection/plot_model_inspection_cross_validation.py

```diff
@@ -0,0 +1,222 @@
+#
+# Once we are happy or at least aware of any model limitations, we can
+# investigate the internals of our model. When we performed the
+# cross-validation, we set the parameter `return_estimator` to `True`.
+#
+# It allows us to get the different predictive models trained and tested within
+# cross-validation.
+
+cv_estimators = cv_results["estimator"]
+
+# %%
+# While the cross-validation allows us to know if our models are reliable, we
+# have not checked if the model at each cross validation fold is similar.
+#
+# Recall that in our case, our baseline model required tuning of the parameter
+# `alpha`. Therefore, up-to-now, we do not know if the optimal `alpha`
+# parameter for all models are similar. In other words, we are asking the
+# question: what is the variance of the `alpha` parameter value across
+# iterations.
+#
+# Indeed, if the `alpha` parameters vary depending on the input data, it
+# might be challenging to put our model in production because we would not be
+# able to fix this hyperparameter.
+#
+# Let's check the `alpha` parameter variance.
+
+alpha = [est[-1].alpha_ for est in cv_estimators]
+plt.hist(alpha, bins=30, density=True)
+plt.xlabel("Alpha")
+plt.ylabel("Density")
+_ = plt.title("Distribution of alpha parameter \nduring cross-validation")
+
+# %%
+# We see that the regularization parameter, `alpha`, values are centered and
+# condensed around 40. This is a good sign and means that most of the models
+# tuned within the cross-validation had similar `alpha` values.
+#
+# However, not only hyperparameter such as `alpha` should be studied. The model
+# parameter coming out of the fitting process should analyzed. In our case, we
+# used a linear model. These models are parametrized and rely on two
+# parameters: `coef_` and `intercept_`. Therefore, we should analyze the
+# variance of these parameters as well.
+#
+# For the sake of simplicity, we are going to solely look at the `coef_`.
+
+import pandas as pd
+
+coefficients = pd.DataFrame([est[-1].coef_ for est in cv_estimators],
+                            columns=X.columns)
+coefficients
+
+# %%
+import seaborn as sns
+
+plt.figure(figsize=(9, 7))
+sns.stripplot(data=coefficients, orient="h", color="k", alpha=0.5)
+sns.boxplot(data=coefficients, orient="h")
+plt.axvline(x=0, color=".5")
+plt.title("Coefficient values our model")
+_ = plt.subplots_adjust(left=0.3)
+
+# %%
+# We observe that the coefficients vary minimally, meaning that all the
+# trained models are similar. Each individual model is expected to more or less
+# give the same predictions.
+#
+# Putting a predictive model in production
+# ----------------------------------------
+#
+# In the above analysis, we saw that the values of lambda did not vary much. We
+# can safely create a model with a fixed `alpha` hyperparameter. Subsequently,
+# we can train the model on the full training set.
+
+from sklearn.linear_model import Ridge
+
+production_model = make_pipeline(StandardScaler(), Ridge(alpha=40))
```

Comment:
Looking at this example again, this line is where I disagree with. Everything up to this point follows logically. From the nested cross validation, we see that the following pipeline:

```python
alphas = np.logspace(-1, 2.5, num=50)
model  = make_pipeline(StandardScaler(), RidgeCV(alphas=alphas))
```

is stable in terms of the coefficients and the r2 score. `model` defines the procedure that produces the stable results. This procedure includes the CV over the `alphas`. When we get to training the production model, I think we should still fit `model` on `X_train`, i.e. not specifying `alpha=40`. Does this logic make sense to you @glemaitre ?

Reply:
Yes, I agree and I think that we should even be extra-careful (I have to read again how I introduce things). After reading https://arxiv.org/pdf/2104.00673.pdf, we should probably be explicit regarding the scores measured via cross-validation and that indeed, they are not reflecting the statistical performance of the refitted model in production.

And as a consequence, there is no reason to fix alpha.